atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/render/StreamStoreRenderer.java

```diff
@@ -558,52 +561,86 @@ private void unmarkStreamsAsUsed() {
             }
 
             private void storeBlocksAndGetFinalMetadata() {
+                String compression = clientSideCompression.compressionType;
                 line("@Override");
                 line("protected StreamMetadata storeBlocksAndGetFinalMetadata(Transaction t, long id, InputStream stream) {"); {
                     line("//Hash the data before compressing it");
                     line("MessageDigest digest = Sha256Hash.getMessageDigest();");
                     line("try (InputStream hashingStream = new DigestInputStream(stream, digest);");
-                    line("        InputStream compressingStream = new LZ4CompressingInputStream(hashingStream)) {"); {
+                    line("        InputStream compressingStream = new " + compression + "(hashingStream)) {"); {
                         line("StreamMetadata metadata = storeBlocksAndGetHashlessMetadata(t, id, compressingStream);");
                         line("return StreamMetadata.newBuilder(metadata)");
                         line("        .setHash(ByteString.copyFrom(digest.digest()))");
                         line("        .build();");
-                    } line("} catch (IOException e) {"); {
-                        line("throw new RuntimeException(e);");
+                    } line("} catch (Exception e) {"); {
+                        line("throw Throwables.rewrapAndThrowUncheckedException(\"Error in creating compression stream\", e);");
```

Comment:
I don't think this change needed to be made - more directly, you shouldn't really be catching Exception.